### PR TITLE
Bug Fix Confidential Client Authentication

### PR DESCRIPTION
**Pull request by joshfischer1108**
# DO NOT MERGE.

## Description
This PR accounts for the work needed to securely authenticate a confidential client with Iridium.

## Motivation and Context

This fixes a bug to allow applications with a secret to authenticate themselves. 

## How has this been tested?
We are still in the process of testing.  Please review the code, and approve if you deem the logic looks good from your perspective.   

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: iridium-core-server/src/main/java/software/iridium/api/TokenAuthenticationFilter.java [hunk 1]
```diff
@@ -56,6 +58,13 @@ protected Object getPreAuthenticatedPrincipal(final HttpServletRequest httpServl
   private String extractBearerToken(final HttpServletRequest request) {
     final var header = request.getHeader(HttpHeaders.AUTHORIZATION);
 
+    if (header != null
+        && header.contains("Basic")
+        && request.getServletPath().equalsIgnoreCase(EXCHANGE_PATH)) {
+      // this is basic authentication
+      return null;
+    }
```
Comment (rwithers) [inline iridium-core-server/src/main/java/software/iridium/api/TokenAuthenticationFilter.java:66]: This snippet bothers me.  It feels like Basic Authentication is called in error here and in my thinking we should log out or throw an error that lets the user know.  But I realize that null obeys the behavior of the interface in all other ways and agrees with the rest of the return values.

File: iridium-core-server/src/test/java/software/iridium/api/TokenAuthenticationFilterTest.java [hunk 2]
```diff
@@ -71,6 +70,23 @@ public void getPreAuthenticatedPrincipal_AllGood_BehavesAsExpected() {
...
+  @Test
+  public void getPreAuthenticatedPrincipal_BasicAuth_BehavesAsExpected() {
+    final var accessTokenValue = "Basic theEncodedTokenValue";
+    final var exchangePath = "/oauth/token";
+
+    when(mockServletRequest.getHeader(eq(HttpHeaders.AUTHORIZATION))).thenReturn(accessTokenValue);
+
+    when(mockServletRequest.getServletPath()).thenReturn(exchangePath);
+
+    assertThat(subject.getPreAuthenticatedPrincipal(mockServletRequest), nullValue());
```
Comment (rwithers) [inline iridium-core-server/src/test/java/software/iridium/api/TokenAuthenticationFilterTest.java:82]: Even the unit test is verifying the null value. . . . If this isn't the path I would expect an error message.

**Review by rwithers — COMMENTED** [review]
Just two comments.

File: iridium-core-server/src/main/java/software/iridium/api/TokenAuthenticationFilter.java (hunk 1 above)
Comment (joshfischer1108) [reply iridium-core-server/src/main/java/software/iridium/api/TokenAuthenticationFilter.java]: Good call, @rwithers .  I felt like I was jamming this logic in incorrectly. 

I've update the PR and rolledback those changes to use a Spring convention.

```
  @Bean
  public WebSecurityCustomizer webSecurityCustomizer() {
    return (web) -> web.ignoring().requestMatchers("/oauth/token");
  }
```

**Pull request merged**